### PR TITLE
Added tests for visibility sampling wrapper

### PR DESCRIPTION
**Pull request by 3vilhamster**
**What changed?**
I've refactored visibility sampling manager:
- Moved it to wrappers
- Separated ratelimiter logic from the manager
- Added tests

There are no logical changes.

**Why?**
Improving test coverage

**How did you test it?**
Unit tests/integration tests.

**Potential risks**

**Release notes**

**Documentation Changes**

### CODE REVIEW

File: common/persistence/wrappers/sampled/visibility_manager.go [hunk 1]
```diff
@@ -44,21 +42,18 @@ const (
 // errPersistenceLimitExceededForList is the error indicating QPS limit reached for list visibility.
 var errPersistenceLimitExceededForList = &types.ServiceBusyError{Message: "Persistence Max QPS Reached for List Operations."}
 
-type visibilitySamplingClient struct {
-	rateLimitersForOpen   *domainToBucketMap
-	rateLimitersForClosed *domainToBucketMap
-	rateLimitersForList   *domainToBucketMap
-	persistence           VisibilityManager
-	config                *SamplingConfig
+type sampledVisibilityManager struct {
```
Comment (taylanisikdemir) [inline common/persistence/wrappers/sampled/visibility_manager.go:45]: nit: already moved this file to `sampled` package so the prefix is redundant and can be dropped from the name. `sampled.sampledVisibilityManager` -> `sampled.visibilityManager`

File: common/persistence/wrappers/sampled/visibility_manager.go [hunk 2]
```diff
@@ -191,157 +162,148 @@ func (p *visibilitySamplingClient) UpsertWorkflowExecution(
...
 }
 
-func (p *visibilitySamplingClient) tryConsumeListToken(domain string) error {
-	rateLimiter := p.rateLimitersForList.getRateLimiter(domain, numOfPriorityForList, p.config.VisibilityListMaxQPS(domain))
+func (p *sampledVisibilityManager) tryConsumeListToken(domain, method string) error {
+	rateLimiter := p.rateLimitersForList.GetRateLimiter(domain)
 	ok, _ := rateLimiter.GetToken(0, 1)
 	if ok {
-		p.logger.Debug("List API request consumed QPS token", tag.WorkflowDomainName(domain), tag.Name(callerFuncName(2)))
+		p.logger.Debug("List API request consumed QPS token", tag.WorkflowDomainName(domain), tag.Name(method))
```
Comment (taylanisikdemir) [inline common/persistence/wrappers/sampled/visibility_manager.go:304]: good idea getting rid of the not-future-safe `callerFuncName` here

File: common/persistence/wrappers/sampled/visibility_manager_test.go [hunk 3]
```diff
@@ -0,0 +1,344 @@
...
+
+	_, err = m.ListOpenWorkflowExecutionsByWorkflowID(context.Background(), &persistence.ListWorkflowExecutionsByWorkflowIDRequest{
+		ListWorkflowExecutionsRequest: persistence.ListWorkflowExecutionsRequest{
+			Domain: testDomain,
+		},
+	})
+	assert.Error(t, err, "second call should fail since underlying call should be blocked by rate limiter")
+}
+
+func TestVisibilityManager_ListClosedWorkflowExecutionsByWorkflowID(t *testing.T) {
```
Comment (taylanisikdemir) [inline common/persistence/wrappers/sampled/visibility_manager_test.go:285]: I know you are not fan of table tests but for 99% identical test function bodies I can't see the point of duplicating so much code. any future change will have to be made in N places. maybe I am missing something here. what is the main reasoning for this way of structuring here?

Comment (3vilhamster) [reply common/persistence/wrappers/sampled/visibility_manager_test.go]: Well, in this case table tests are not easy, setup is complicated, though assertion is similar.
I gave it a try and wrote it in table tests style. Give it a look.

**Review by taylanisikdemir — APPROVED** [review]

File: common/persistence/wrappers/sampled/visibility_manager_test.go (hunk 3 above)
Comment (taylanisikdemir) [reply common/persistence/wrappers/sampled/visibility_manager_test.go]: looks good! thanks

**Pull request merged**